### PR TITLE
Minor Silicon PDA Improvements

### PR DESCRIPTION
**Pull request by Nexusuxen**
[QOL][BUG - MINOR]

## About the PR 

General improvements are as follows:
AI PDA now starts in the Party Line and silicon groups
Silicon group is now correctly set to private
Cyborg death alerts will also be sent to the silicon frequency
AIs now have 5 new messenger groups for the following mailbots: Health, Security (Securitron arrests, creating arrest status via records, and security scanner alerts), Cloning Alerts
(The AI is able to effectively mute mailbots by toggling on/off their respective group)

The group names for the mailbots are honestly kinda bad, so if anyone comes up with better ones please let me know and I'd be happy to change them!
There are certain fringe cases that I was unable to test in singleplayer and due to my lack of experience with admin tools, such as cloning alerts showing up properly. 

## Why's this needed? 

The AI being able to see and, importantly, toggle on/off the receiving of certain mailbots can provide pretty dang useful information
The AI should also 100% start in the silicon group since, well, it's a silicon, and the machine translator implant exists, so snooping on rogue AIs and borgs is a bit harder
The AI must also suffer the pain everyone else feels when someone uses Party Line for the first time in a round

## Changelog

```
(u)Aft2001:
(*)AIs can now see certain mailbot alerts and effectively toggle them by muting their respective groups in their messenger
(+)The silicon PDA group is now actually private, and the AI starts in it
(+)Borg death alerts now also send to the silicon PDA group
(+)The AI now starts in the Party Line group
```

### CODE REVIEW

**Comment by Tiggersaurus** [comment]
Yessssss

File: code/mob/living/silicon/robot.dm [hunk 1]
```diff
@@ -226,7 +226,6 @@
 			logTheThing("combat", src, null, "'s AI controlled cyborg body was destroyed [log_health(src)] at [log_loc(src)].") // Brought in line with carbon mobs (Convair880).
 			src.mainframe.return_to(src)
 		setdead(src)
-		borg_death_alert()
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:229]: ?

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -2763,56 +2762,58 @@
...
+			playsound(get_turf(src), "sound/impact_sounds/Metal_Hit_Light_1.ogg", 40, 1)
+			if (istype(src.loc,/turf/)) make_cleanable(/obj/decal/cleanable/robot_debris, src.loc)
+			elecflash(src,power = 2)
+
+			if (istype(part,/obj/item/parts/robot_parts/chest/))
+				src.visible_message("<b>[src]'s</b> chest unit is destroyed!")
+				src.part_chest = null
+			if (istype(part,/obj/item/parts/robot_parts/head/))
+				src.visible_message("<b>[src]'s</b> head breaks apart!")
+				//borg_death_alert()//no head means you dead
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:2779]: ?

File: code/mob/living/silicon/robot.dm [hunk 3]
```diff
@@ -2763,56 +2762,58 @@
...
+					src.visible_message("<b>[src]'s</b> legs are destroyed!")
+					src.part_leg_r = null
+					src.part_leg_l = null
+				if (part.slot == "leg_left")
+					src.visible_message("<b>[src]'s</b> left leg breaks off!")
+					src.part_leg_l = null
+				if (part.slot == "leg_right")
+					src.visible_message("<b>[src]'s</b> right leg breaks off!")
+					src.part_leg_r = null
+			//var/loseslot = part.slot //ZeWaka: Fix for null.slot
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:2810]: del

File: code/mob/living/silicon/robot.dm [hunk 4]
```diff
@@ -2763,56 +2762,58 @@
...
+			if(part.robot_movement_modifier)
+				REMOVE_MOVEMENT_MODIFIER(src, part.robot_movement_modifier, part.type)
+			src.update_bodypart()
+			//src.update_bodypart(loseslot)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:2814]: del

File: code/modules/robotics/bot/secbot.dm [hunk 5]
```diff
@@ -955,16 +955,26 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
...
+				var/datum/signal/pdaSignal = get_free_signal()
+				pdaSignal.source = src
+				pdaSignal.data["command"] = "text_message"
+				pdaSignal.data["sender_name"] = "SECURITY-MAILBOT"
+				pdaSignal.data["message"] = "[message2send]"
+
+				pdaSignal.data["address_1"] = "00000000"
+				pdaSignal.data["group"] = mailgroup
+				pdaSignal.data["sender"] = "00000000"
+				//pdaSignal.data = list("address_1"="00000000", "command"="text_message", "sender_name"="SECURITY-MAILBOT",  "group"=MGD_SECURITY, "sender"="00000000", "message"="[message2send]")
```
Comment (ZeWaka) [inline code/modules/robotics/bot/secbot.dm:974]: del

File: code/modules/robotics/bot/secbot.dm [hunk 6]
```diff
@@ -1137,19 +1147,28 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
...
+				for(var/mailgroup in mailgroups)
+					var/datum/signal/pdaSignal = get_free_signal()
+					pdaSignal.data["command"] = "text_message"
+					pdaSignal.data["sender_name"] = "SECURITY-MAILBOT"
+					pdaSignal.data["message"] = "[message2send]"
+
+					pdaSignal.data["address_1"] = "00000000"
+					pdaSignal.data["group"] = mailgroup
+					pdaSignal.data["sender"] = "00000000"
+				//pdaSignal.data = list("address_1"="00000000", "command"="text_message", "sender_name"="SECURITY-MAILBOT",  "group"=MGD_SECURITY, "sender"="00000000",\
```
Comment (ZeWaka) [inline code/modules/robotics/bot/secbot.dm:1167]: del

File: code/obj/machinery/secscanner.dm [hunk 7]
```diff
@@ -61,15 +61,29 @@
...
+					pdaSignal.data["command"] = "text_message"
+					pdaSignal.data["sender_name"] = "SECURITY-MAILBOT"
+					pdaSignal.data["message"] = "Notification: An item [I.name] failed checkpoint scan at [scan_location]! Threat Level : [contraband]"
+
+					pdaSignal.data["address_1"] = "00000000"
+					pdaSignal.data["group"] = mailgroup
+					pdaSignal.data["sender"] = "00000000"
+
+					//pdaSignal.data = list("address_1"="00000000", "command"="text_message", "sender_name"="SECURITY-MAILBOT",  "group"=MGD_SECURITY, "sender"="00000000", "message"="Notification: An item [I.name] failed checkpoint scan at [scan_location]! Threat Level : [contraband]")
+					//pdaSignal.transmission_method = TRANSMISSION_RADIO
```
Comment (ZeWaka) [inline code/obj/machinery/secscanner.dm:84]: del

File: code/obj/machinery/secscanner.dm [hunk 8]
```diff
@@ -112,11 +126,21 @@
...
+							for(var/mailgroup in mailgroups)
+								var/datum/signal/pdaSignal = get_free_signal()
+								pdaSignal.data["command"] = "text_message"
+								pdaSignal.data["sender_name"] = "SECURITY-MAILBOT"
+								pdaSignal.data["message"] = "NOTIFICATION: [uppertext(perpname)] FAILED A VIBE CHECK AT [uppertext(scan_location)]! BAD VIBES LEVEL : [contraband]"
+
+								pdaSignal.data["address_1"] = "00000000"
+								pdaSignal.data["group"] = mailgroup
+								pdaSignal.data["sender"] = "00000000"
+								//pdaSignal.data = list("address_1"="00000000", "command"="text_message", "sender_name"="SECURITY-MAILBOT",  "group"=MGD_SECURITY, "sender"="00000000", "message"="NOTIFICATION: [uppertext(perpname)] FAILED A VIBE CHECK AT [uppertext(scan_location)]! BAD VIBES LEVEL : [contraband]")
```
Comment (ZeWaka) [inline code/obj/machinery/secscanner.dm:140]: del

File: code/obj/machinery/secscanner.dm [hunk 9]
```diff
@@ -151,14 +175,24 @@
...
+							pdaSignal.source = src
+							pdaSignal.transmission_method = TRANSMISSION_RADIO
+							pdaSignal.data["command"] = "text_message"
+							pdaSignal.data["sender_name"] = "SECURITY-MAILBOT"
+							pdaSignal.data["message"] = "Notification: [perpname] failed checkpoint scan at [scan_location]! Threat Level : [contraband]"
+
+							pdaSignal.data["address_1"] = "00000000"
+							pdaSignal.data["group"] = mailgroup
+							pdaSignal.data["sender"] = "00000000"
+							//pdaSignal.data = list("address_1"="00000000", "command"="text_message", "sender_name"="SECURITY-MAILBOT",  "group"=MGD_SECURITY, "sender"="00000000", "message"="Notification: [perpname] failed checkpoint scan at [scan_location]! Threat Level : [contraband]")
```
Comment (ZeWaka) [inline code/obj/machinery/secscanner.dm:192]: del

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by Nexusuxen** [comment]
Merge conflict fixed, fixing commented out code; will add reasoning for deletion of borg death proc call after

File: code/mob/living/silicon/robot.dm (hunk 1 above)
Comment (Nexusuxen) [reply code/mob/living/silicon/robot.dm:229]: Completely forgot about that bug fix; in certain circumstances, the borg death alert proc can be called more than once, resulting in weird behavior with double PDA messages (extensive testing and headscratching with people in #ImCoder indicated that this was the most likely cause; deleting this line fixed the issue and HOPEFULLY won't fuck other things over)

File: code/mob/living/silicon/robot.dm [hunk 10]
```diff
@@ -2317,7 +2316,7 @@
 
 	proc/borg_death_alert(modifier = ROBOT_DEATH_MOD_NONE)
 		var/message = null
-		var/mailgroup = MGD_MEDRESEACH
+		var/list/mailgroups = list(MGD_MEDRESEACH, "silicon")
```
Comment (mordent-goonstation) [inline code/mob/living/silicon/robot.dm:2319]: We should probably use constants for new mailgroups (i.e. make one for `"silicon"`), I think?

File: code/mob/living/silicon/robot.dm [hunk 11]
```diff
@@ -2333,7 +2332,8 @@
 			else	//Someone passed us an unkown modifier
 				message = "UNKNOWN ERROR: [src] in [myarea]"
 
-		if (message && mailgroup && radio_connection)
+		for(var/mailgroup in mailgroups)
+			//if (message && mailgroup && radio_connection)
```
Comment (mordent-goonstation) [inline code/mob/living/silicon/robot.dm:2336]: Clean up comment. I also think you might still want the `if (message)` check before the for-loop (please check, packet code is not my strong suit).

File: code/modules/networks/computer3/sec_rec.dm [hunk 12]
```diff
@@ -734,18 +734,19 @@
...
 			//Create a PDA mass-message string.
-			signal.data["command"] = "text_message"
-			signal.data["sender_name"] = "SEC-MAILBOT"
-			signal.data["group"] = src.setup_mailgroup //Only security PDAs should be informed.
-			signal.data["message"] = "Alert! Crewman \"[perp_name]\" has been flagged for arrest by [src.authenticated]!"
-
-			src.log_string += "<br>Arrest notification sent."
-			last_arrest_report = world.time
-			peripheral_command("transmit", signal, "\ref[src.radiocard]")
+			var/list/mailgroups = list(MGD_SECURITY, "AI-Sec_Records Alerts")
```
Comment (mordent-goonstation) [inline code/modules/networks/computer3/sec_rec.dm:738]: MGD constant?

File: code/modules/robotics/bot/secbot.dm [hunk 13]
```diff
@@ -958,16 +958,26 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 		if (report_arrests)
 			var/bot_location = get_area(src)
 			var/datum/radio_frequency/transmit_connection = radio_controller.return_frequency(FREQ_PDA)
-			var/datum/signal/pdaSignal = get_free_signal()
 			var/message2send
+			var/list/mailgroups = list(MGD_SECURITY, "AI-Arrest Alerts")
```
Comment (mordent-goonstation) [inline code/modules/robotics/bot/secbot.dm:962]: MGD constant? A few other places too, can't be bothered to comment them all.

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

**Comment by ZeWaka** [comment]
I'd like to see this merged if the requests can be addressed.

File: code/mob/living/silicon/robot.dm [hunk 14]
```diff
@@ -2333,7 +2333,7 @@
 				message = "UNKNOWN ERROR: [src] in [myarea]"
 
 		for(var/mailgroup in mailgroups)
-			//if (message && mailgroup && radio_connection)
+			if (message && mailgroup && radio_connection)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:2336]: indentation

**Pull request closed without merging**